skalenetwork/skaled
# Docs Development Automation Added

Reason for Update
---
Adds "hot reload/watch" functionality when updating documentation

Updates
---
- Added submodule doc-ui to root which calls: https://github.com/skalenetwork/doc-ui
- Added doc-utils as npm package (GitHub Package) -> https://github.com/skalenetwork/doc-utils
- Added local-playbook-ui.yml -> Core Hot Reload Template for Live Doc Edits

How to Test
---
- Checkout this PR, and setup as normal ```yarn```
- Run ```yarn docs:watch``` and wait for the UI to build, once built opening ```127.0.0.1:444``` in your browser should allow you to select the **skaled** folder, you must manually click into the file
- After manually navigating in the browser changes will occur in the browser automatically when changes are made in the **docs** folder. 
- See notes below

Notes
---
- Due to minor docgen, this repo may show _errors_ when building, however, that won't impact the built and doesn't slow down reload time much
